release.go

```diff
@@ -242,27 +248,43 @@ func buildReleaseCommits(commits []Commit, cfg ReleaseConfig) []ReleaseCommit {
 
 		c := ReleaseCommit{
 			Commit:       commit,
-			ReleaseNote:  determineCommitReleaseNote(commit, cfg.ReleaseNoteGenerator.UseReleaseNoteBlock),
+			ReleaseNote:  extractReleaseNote(commit.Subject, commit.Body, gen.UseReleaseNoteBlock),
 			CategoryName: determineCommitCategory(commit, mergeCommits[commit.Hash], cfg.CommitCategories),
 		}
+
+		if gen.UseDefaultMergeCommitParser {
+			prNumber, ok := commit.PullRequestNumber()
+			if !ok {
+				continue
+			}
+
+			pr, err := ghClient.getPullRequest(ctx, event.Owner, event.Repo, prNumber)
```

Comment:
How about using https://docs.github.com/en/rest/reference/pulls#list-pull-requests to avoid rate-limit error?

We can use ListPullRequests to fetch a large number of closed PR (e.g. 1000) at the first time (this costs 1 call) to check them at local.
All missing ones can be requested directly by GetPullRequest.

Reply:
Oh, that's pretty good. Let me apply it.